src/net_processing.cpp

```diff
@@ -2043,23 +2087,37 @@ static bool ProcessGetCFCheckPt(CNode* pfrom, CDataStream& vRecv, const CChainPa
         return false;
     }
 
-    bool index_in_sync = false;
     std::vector<uint256> headers(stop_index->nHeight / CFCHECKPT_INTERVAL);
+    {
+        boost::shared_lock<boost::shared_mutex> _lock(active_chain_cf_headers_mtx);
```

Comment:
To be clear: at the moment, the `shared_lock` is more or less moot because this is only ever called into from a single thread (`msghand`). I'm not necessarily opposed to using the shared_lock before necessary, but might be worth falling back to our standard `RecursiveMutex` (which comes with lock diagnostics) until and if we actually hit this from multiple threads.

Reply:
Oh, for some reason I thought there were multiple message handler threads. Since there's just one and this doesn't require recursive locking, I think going to a `std::mutex` makes sense. Agree?

Reply:
@jimpo yep, that'd be my inclination.